public/components/integrations/components/integration.tsx

```diff
@@ -0,0 +1,326 @@
+        }
+        return parsedResponse.data;
+      })
+      .then((assets) => setAssets(assets))
+      .catch((err: any) => {
+        console.error(err.message);
+      });
+  }, [integration]);
+
+  async function addIntegrationRequest(
+    addSample: boolean,
+    templateName: string,
+    name?: string,
+    dataSource?: string
+  ) {
+    setLoading(true);
+    if (addSample) {
+      createDataSourceMappings(`ss4o_${integration.type}-${integrationTemplateId}-*-sample`);
+      name = `${integrationTemplateId}-sample`;
+      dataSource = `ss4o_${integration.type}-${integrationTemplateId}-sample-sample`;
+    }
+
+    const response: boolean = await http
+      .post(`${INTEGRATIONS_BASE}/store/${templateName}`, {
+        body: JSON.stringify({ name, dataSource }),
+      })
+      .then((_res) => {
+        setToast(`${name} integration successfully added!`, 'success');
+        window.location.hash = `#/installed/${_res.data?.id}`;
+        return true;
+      })
+      .catch((_err) => {
+        setToast(
+          'Failed to load integration. Check Added Integrations table for more details',
+          'danger'
+        );
+        return false;
+      });
+    if (!addSample || !response) {
+      setLoading(false);
+      return;
+    }
+    const data: { sampleData: unknown[] } = await http
+      .get(`${INTEGRATIONS_BASE}/repository/${templateName}/data`)
+      .then((res) => res.data)
+      .catch((err) => {
+        console.error(err);
+        setToast('The sample data could not be retrieved', 'danger');
+        return { sampleData: [] };
+      });
+    const requestBody =
+      data.sampleData
+        .map((record) => `{"create": { "_index": "${dataSource}" } }\n${JSON.stringify(record)}`)
+        .join('\n') + '\n';
+    fetch(`/api/console/proxy?path=${dataSource}/_bulk&method=POST`, {
+      method: 'POST',
+      body: requestBody,
+      headers: [
+        ['osd-xsrf', 'true'],
+        ['Content-Type', 'application/json; charset=utf-8'],
+      ],
+    })
+      .catch((err) => {
+        console.error(err);
+        setToast('Failed to load sample data', 'danger');
+      })
+      .finally(() => {
+        setLoading(false);
+      });
+  }
+
+  const tabs = [
+    {
+      id: 'assets',
+      name: 'Asset List',
+      disabled: false,
+    },
+    {
+      id: 'fields',
+      name: 'Integration Fields',
+      disabled: false,
+    },
+  ];
+
+  const [selectedTabId, setSelectedTabId] = useState('assets');
+
+  const onSelectedTabChanged = (id) => {
+    setSelectedTabId(id);
+  };
+
+  const renderTabs = () => {
+    return tabs.map((tab, index) => (
+      <EuiTab
+        onClick={() => onSelectedTabChanged(tab.id)}
+        isSelected={tab.id === selectedTabId}
+        disabled={tab.disabled}
+        key={index}
+        data-test-subj={tab.id}
+      >
+        {tab.name}
+      </EuiTab>
+    ));
+  };
+
+  if (Object.keys(integration).length === 0) {
```

Comment:
This might be easier to read/follow/maintain using React Suspense
https://react.dev/reference/react/Suspense

Create a follow-on task to cover this on different PR